src/game/Spell.cpp

```diff
@@ -5100,9 +5100,7 @@ SpellCastResult Spell::CheckCast(bool strict)
                     else
                     {
                         Pet* dbPet = new Pet;
-                        if (dbPet->LoadPetFromDB((Player*)m_caster, 0))
-                            return SPELL_CAST_OK;           // still returns an error to the player, so this error must come from somewhere else...
-                        else
+                        if (!dbPet->TryLoadFromDB((Player*)m_caster, 0))
```

Comment:
Still can't have hunters going through the summon function ever, you have to return them in their own function (take a look at how it was done back in april, before any of my work - hunter's pets have always been returned out of summonPet early as their pets are not created by that function, only summoned via it, hunter's pets are created through Tamebeast or whatever it's called -> most of what is in SummonPet is for creating the pet, this is why warlocks only go through it once with each pet.